### PR TITLE
PACT-604 Create a List Legal Status API Service using stubbed data

### PR DESCRIPTION
**Pull request by userjd**

### CODE REVIEW

**Review by rwalpole — CHANGES_REQUESTED** [review]
There is a part missing here which is the headers for the reply message. In the AsyncAPI schema these are described in the `AssetLegalStatusSummaryList` and the missing headers are as follows:
- `OMGApplicationID` - this will be `PACS001` for ctd-omega-services
- `OMGMessageTypeID` - this will be `ODLISALS001` for the legal status summary reply
- `OMGMessageFormat` - this will be `application/json` for all reply messages currently
These will need to be added in the `LocalProducer.send` function. We already add the appropriate headers in the `LocalProducer.sendError` function so it will look very similar to that.

**Comment by userjd** [comment]
Done. Thanks @rwalpole

File: src/main/scala/uk/gov/nationalarchives/omega/api/business/echo/EchoService.scala [hunk 1]
```diff
@@ -27,15 +27,15 @@ class EchoService extends BusinessService with BusinessRequestValidation {
...
-      request.text.trim.nonEmpty,
+      request.text.nonEmpty,
       request,
       NonEmptyChain.one(TextIsNonEmptyCharacters("Echo Text cannot be empty."))
     )
 
   override def process(request: BusinessServiceRequest): Either[BusinessServiceError, BusinessServiceReply] =
-    if (request.text.contains("ERROR")) {
-      Left(EchoExplicitError(s"Explicit error: ${request.text}"))
+    if (request.text.get.contains("ERROR")) {
```
Comment (rwalpole) [inline src/main/scala/uk/gov/nationalarchives/omega/api/business/echo/EchoService.scala:36]: Using `get` could throw a `NoSuchElementException` if it is `None`. In this case you don't need to change anying as `contains` [can be used on an Option](https://www.scala-lang.org/api/current/scala/Option.html#contains[A1%3E:A](elem:A1):Boolean).

File: src/main/scala/uk/gov/nationalarchives/omega/api/business/echo/EchoService.scala [hunk 2]
```diff
@@ -27,15 +27,15 @@ class EchoService extends BusinessService with BusinessRequestValidation {
...
+      Left(EchoExplicitError(s"Explicit error: ${request.text.get}"))
     } else {
-      Right(EchoReply(s"The Echo Service says: ${request.text}"))
+      Right(EchoReply(s"The Echo Service says: ${request.text.get}"))
```
[the remark this answers is not in the document]
Comment (userjd) [reply src/main/scala/uk/gov/nationalarchives/omega/api/business/echo/EchoService.scala]: Done

File: src/main/scala/uk/gov/nationalarchives/omega/api/business/echo/EchoService.scala (hunk 1 above)
Comment (userjd) [reply src/main/scala/uk/gov/nationalarchives/omega/api/business/echo/EchoService.scala]: Added `nonEmpty` check

File: src/main/scala/uk/gov/nationalarchives/omega/api/messages/StubData.scala [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package uk.gov.nationalarchives.omega.api.messages
+
+import uk.gov.nationalarchives.omega.api.models.LegalStatus
+
+class StubDataImpl extends StubData
+
+trait StubData {
```
[the remark this answers is not in the document]
Comment (userjd) [reply src/main/scala/uk/gov/nationalarchives/omega/api/messages/StubData.scala:28]: As discussed not changed as being used for testing by `TestStubData`

**Review by rwalpole — APPROVED** [review]

**Pull request merged**